editor/components/post-preview-button/index.js

```diff
@@ -51,25 +49,36 @@ export class PostPreviewButton extends Component {
 	}
 
 	saveForPreview( event ) {
-		const { isDirty, isNew } = this.props;
+		const { isAutosaveable, previewLink } = this.props;
 
-		// Let default link behavior occur if no changes to saved post
-		if ( ! isDirty && ! isNew ) {
+		// If there are no changes to autosave, we cannot perform the save, but
+		// if there is an existing preview link (e.g. previous published post
+		// autosave), it should be reused as the popup destination.
+		if ( ! isAutosaveable && ! previewLink ) {
 			return;
 		}
 
-		// Save post prior to opening window
-		this.props.autosave();
-
 		// Open a popup, BUT: Set it to a blank page until save completes. This
 		// is necessary because popups can only be opened in response to user
 		// interaction (click), but we must still wait for the post to save.
 		event.preventDefault();
 		this.previewWindow = window.open(
-			'about:blank',
+			previewLink ? previewLink : 'about:blank',
```

Comment:
In this case, `previewLink || 'about:blank'` might be a bit nicer than the ternary, but it's just a style thing really.

Reply:
>In this case, `previewLink || 'about:blank'` might be a bit nicer than the ternary, but it's just a style thing really.

I'd agree, though to the general point this sort of clever use of `||` and `&&` can sometimes bite when people think only of the fallback case, but not the fall...-forward (?) case, i.e. not anticipating the evaluation to the falsey value as actually being the assignment.

```js
// false || null
null

// 0 && null
0
```

The latter one is a common pain-point in React where people have the tendency to do things like:

```js
array.length && (
    <ul>{ /* ... */ }</ul>
)
```

...not anticipating that when the array is empty, there will be the `0` as output.

Demo: https://codepen.io/aduth/pen/dWwzrL

Reply:
Heh, true I guess. I wouldn't ever think to use `array.length && ( […] )` but if others do: fair enough!